<?xml version="1.0"?>
<!--
     LAUNCH-FILE for
     RMS3xx - Radar device
     EXPERIMENTAL - brand new driver
     Only SOPAS-ASCII supported.
  -->
<launch>
    <!-- x y z qx qy qz qw frame_id child_frame_id  period_in_ms-->
    <!--
    <node pkg="tf" type="static_transform_publisher" name="Radar_map_broadcaster" args="0 0 0 0 0 0 1 /Radar /map 100" />

    <node pkg="tf" type="static_transform_publisher" name="laser_map_broadcaster" args="0 0 0 0 0 0 1 /laser /map 100" />

    <node pkg="tf" type="static_transform_publisher" name="Object_map_broadcaster" args="0 0 0 0 0 0 1 /Objects /map 100" />
    -->
    <node name="sick_rms_3xx" pkg="sick_scan" type="sick_generic_caller" respawn="false" output="screen">
        <!-- default values: -->
        <param name="scanner_type" type="string" value="sick_rms_3xx"/>
        <param name="range_max" type="double" value="25.0"/>
        <param name="hostname" type="string" value="192.168.0.232"/>
        <param name="port" type="string" value="2112"/>
        <param name="timelimit" type="int" value="5"/>
        <!-- tracking_mode 0: BASIC-Tracking - use for tracking smaller objects -->
        <!-- tracking_mode 1: TRAFFIC-Tracking - use for tracking larger objects like vehicles -->
        <param name="tracking_mode" type="int" value="0"/>
        <!-- raw_targets: switch transfer of radar raw targets on or off  -->
        <!-- Raw targets are input for the internal tracking algorithm    -->
        <!-- and can be used to develop your own radar tracking algorithm -->
        <param name="transmit_raw_targets" type="bool" value="true"/>
        <param name="trnasmit_objects" type="bool" value="true"/>
        <param name="emul_sensor" type="bool" value="false"/>
    </node>

    <node name="sick_mrs_6xxx" pkg="sick_scan" type="sick_generic_caller" respawn="false" output="screen">
        <!-- default values: -->
        <!--
          <param name="scanner_type" type="string" value="sick_mrs_6xxx"
          <param name="min_ang" type="double" value="-2.35619449019" />
          <param name="max_ang" type="double" value="2.35619449019" />
          <param name="intensity" type="bool" value="True" />
          <param name="skip" type="int" value="0" />
          <param name="frame_id" type="str" value="laser" />
          <param name="time_offset" type="double" value="-0.001" />
          <param name="publish_datagram" type="bool" value="False" />
          <param name="subscribe_datagram" type="bool" value="false" />
          <param name="device_number" type="int" value="0" />
          <param name="range_min" type="double" value="0.05" />
        -->
        <param name="scanner_type" type="string" value="sick_mrs_6xxx"/>
        <param name="range_max" type="double" value="25.0"/>
        <param name="hostname" type="string" value="192.168.0.24"/>
        <param name="port" type="string" value="2112"/>
        <param name="timelimit" type="int" value="5"/>
        <param name="min_ang" type="double" value="-1.047197333"/>
        <param name="max_ang" type="double" value="+1.047197333"/>
    </node>

    <node name="radar_object_marker" pkg="sick_scan" type="sick_scan_radar_object_marker" respawn="false"
          output="screen"/>

</launch>
